troubleshooting/debugging.md

```diff
@@ -642,6 +654,55 @@ Path                /logs/request.log
 
 In other cases, the original config would be correct. For example, assume your app produces first a log file `request.log` and then next creates and writes new logs to a new file named `request.log1` and then next `request.log2`. And then when the max number of files are reached, the original `request.log` is deleted and then a new `request.log` is written. If that is the scenario, then the original Path pattern with a wildcard at the end is correct since it is necessary to match all log files where new logs are written. This method of log rotation is the behavior of the [log4j DirectWriteRolloverStrategy](https://logging.apache.org/log4j/2.x/manual/appenders.html#DirectWriteRolloverStrategy) where there is no file renaming. 
 
+### Best Practices
+
+*Configuration changes which either help mitigate/prevent common issues, or help in debugging issues.*
+
+#### Set Aliases on all Inputs and Outputs
+
+Fluent Bit supports [configuring an alias name](https://docs.fluentbit.io/manual/administration/monitoring#configuring-aliases) for each input and output. 
+
+The Fluent Bit internal metrics and most error/warn/info/debug log messages use either the alias or the Fluent Bit defined input name for context. The Fluent Bit input name is the plugin name plus an ID number; for example if you have 4 tail inputs you would have `tail.1`, `tail.2`, `tail.3`, `tail.4`. The numbers are assigned based on the order in your configuration file. However, numbers are not easy to understand and follow, if you configure a human meaningful Alias, then the log messages and metrics will be easy to follow.
+
+Example input definition with Alias:
+
+```
+[INPUT]
+    Name  tail
+    Alias ServiceMetricsLog
+    ...
+```
+
+Example output definition with Alias:
+
+```
+[OUTPUT]
+    Name cloudwatch_logs
+    Alias SendServiceMetricsLog
+```
+
+#### Tail Config with Best Practices
+
+First take a look at the sections of this guide specific to tail input:
+* [Tail Input Skipping File](#tail-input-skipping-file)
+* [Always use multiline in the tail input](#always-use-multiline-in-the-tail-input)
+* [Tail input duplicates logs because of rotation](#tail-input-duplicates-logs-because-of-rotation)
+* [Tail Ignore_Older](#tail-input-ignore_older)
+* [Tail Permission Errors](#tail-permission-errors)
+
+
+A Tail configuration following the above best practices might look like:
+
+```
+[INPUT]
+    Name tail
```

Comment:
In the same PR that I added the alias best practice, I forgot to add an Alias in the tail best practice example :(